### ISSUE TITLE
Max to Crashes, No Notification Tray Icon and Relaunching Doesn't Help

### ISSUE THREAD

**Issue by SapneshNaik** [digitalcreations/MaxTo#618]
**Describe the bug**

I'm noticing that MaxTo disappears from the Notification tray area, and when I relauch it, the app crashes with a message saying MaxTo has crashes and will now try to restart itself. And even after restarting (UAC for MaxTo.Core) I see no MaxTo Icon in the tray area. The only way to bring it back now is to go into task manager, kill all MaxTo related processes (Core, Server, Companion, etc.) and then relaunch MaxTo again. But this happens often enough that it is annoying.

**To Reproduce**
I've not been able to figure out the exact steps to reproduce this problem, but I have seen this happen mostly when my laptop comes back from Sleep mode and when I disconnect and reconnecrt  an external monitor.

**Screenshots**

![MaxTO error](https://user-images.githubusercontent.com/26362823/93892374-190cf000-fd0a-11ea-9577-599b10d8b9c7.png)

**System information:**
- Windows version: Version 1909 (OS Build 18363.1082)
- MaxTo version: 2.1.0-beta3

**Additional context**

[logs.zip](https://github.com/digitalcreations/MaxTo/files/5261956/logs.zip)

**Comment by vegardlarsen** [comment]
@SapneshNaik Can you get me your runtime logs? See `%AppData%\MaxTo` and look for any log files on the dates where you experience crashes.

**Comment by SapneshNaik** [comment]
@vegardlarsen the attached logs.zip in the issue has the latest log lige along with the config file and the monitor files. However, I'm also attaching logs for all crashes with this comment

[all_logs.zip](https://github.com/digitalcreations/MaxTo/files/5262058/all_logs.zip)

**Comment by vegardlarsen** [comment]
Sorry I missed that. I can confirm that this is a know issues (actually the most frequent crash report for version 2.1.0-beta.3):

```
2020-09-22 11:31:41 [UserInterface@2.1.0.675] [Fatal] [] Unhandled exception "TargetInvocationException" at "   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Caliburn.Micro.XamlPlatformProvider.<>c__DisplayClass13_3.<GetViewCloseAction>b__1()
   at Caliburn.Micro.XamlPlatformProvider.OnUIThread(Action action)
   at Caliburn.Micro.Screen.TryClose(Nullable`1 dialogResult)
   at MaxTo.UI.ViewModels.SetRegionsWindowManager.<Activate>b__11_1(Object sender, DeactivationEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at Caliburn.Micro.Screen.Caliburn.Micro.IDeactivate.Deactivate(Boolean close)
   at Caliburn.Micro.WindowManager.WindowConductor.Closed(Object sender, EventArgs e)
   at System.EventHandler.Invoke(Object sender, EventArgs e)
   at System.Windows.Window.OnClosed(EventArgs e)
   at System.Windows.Window.WmDestroy()
   at System.Windows.Window.WindowFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.PublicHooksFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)"
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: Cannot set Visibility to Visible or call Show, ShowDialog, Close, or WindowInteropHelper.EnsureHandle while a Window is closing.
   at System.Windows.Window.VerifyNotClosing()
   at System.Windows.Window.InternalClose(Boolean shutdown, Boolean ignoreCancel)
   at System.Windows.Window.Close()
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Caliburn.Micro.XamlPlatformProvider.<>c__DisplayClass13_3.<GetViewCloseAction>b__1()
   at Caliburn.Micro.XamlPlatformProvider.OnUIThread(Action action)
   at Caliburn.Micro.Screen.TryClose(Nullable`1 dialogResult)
   at MaxTo.UI.ViewModels.SetRegionsWindowManager.<Activate>b__11_1(Object sender, DeactivationEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at Caliburn.Micro.Screen.Caliburn.Micro.IDeactivate.Deactivate(Boolean close)
   at Caliburn.Micro.WindowManager.WindowConductor.Closed(Object sender, EventArgs e)
   at System.EventHandler.Invoke(Object sender, EventArgs e)
   at System.Windows.Window.OnClosed(EventArgs e)
   at System.Windows.Window.WmDestroy()
   at System.Windows.Window.WindowFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.PublicHooksFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
```

We will fix this for 2.1.0-beta.4.

**Comment by vegardlarsen** [comment]
This has been fixed for 2.1.0-beta.4.
